test/action_view/component_test.rb

```diff
@@ -365,6 +365,30 @@ def test_render_check
     assert_empty render_inline(RenderCheckComponent).text, ""
   end
 
+  def test_disable_validation
+    component_without_validation = WithoutValidationComponent.new
+
+    assert_includes component_without_validation.render_in(controller.view_context), "Rendered"
+
+    WithoutValidationComponent.with_validation!
+
+    component_with_validation = WithoutValidationComponent.new
+
+    exception = assert_raises ActiveModel::ValidationError do
+      component_without_validation.render_in(controller.view_context)
+    end
+    assert_equal exception.message, "Validation failed: Required can't be blank"
+  end
+
+  def test_dry_validation
+    assert_includes render_inline(DryValidationComponent, required: "text").text, "Rendered"
+
+    exception = assert_raises RuntimeError do
+      render_inline(DryValidationComponent)
+    end
+    assert_equal exception.message, "required: must be filled"
+  end
```

Comment:
Is it important to test dry-validation here? The real test here is that "if I override `validate!` and throw exceptions those exceptions are not rescued` We could do that with any kind of override, there is nothing unique about the fact that it works with dry-validation (which would give us one less dev dependency)

Reply:
I think it is better to have real world example here